rusty-machine/src/linalg/matrix/iter.rs

```diff
@@ -0,0 +1,194 @@
+use std::marker::PhantomData;
+use std::slice;
+
+use super::{Matrix, MatrixSlice, MatrixSliceMut};
+
+#[derive(Debug)]
+pub struct Rows<'a, T: 'a> {
+    row_start: *const T,
+    row_pos: usize,
+    slice_rows: usize,
+    slice_cols: usize,
+    row_stride: isize,
+    _marker: PhantomData<&'a T>,
+}
+
+/// Iterates over the matrix data by slice
+impl<'a, T> Iterator for Rows<'a, T> {
+    type Item = &'a [T];
```

Comment:
I think I could imagine wanting the iterator elements to each be a `MatrixSlice<T>` or a `Vector<T>` rather than a `&[T]`? I'm not sure.